comp/otelcol/converter/impl/autoconfigure.go

```diff
@@ -39,6 +43,9 @@ func enhanceConfig(conf *confmap.Conf) {
 
 	// datadog connector
 	changeDefaultConfigsForDatadogConnector(conf)
+
+	// add datadog agent sourced config
```

Comment:
cc @mackjmr you may want to mention this in https://github.com/DataDog/datadog-agent/pull/28583 too

Reply:
@songy23 will do, just to make sure I understand this code:
If:
- no `api_key` is set in the collector config OR
- the string in `api_key` in the collector config is a secret

Then we will pull the api key from the agent config and use that, as well as set the site to what is in the agent config if site is not already set.

Does that sound about right ?

Reply:
Yes, it also applies to when api_key is explicitly set to an empty string

Reply:
Thanks, added in this commit: https://github.com/DataDog/datadog-agent/pull/28583/commits/0637a7fea17dc76fee3bd37845d0e97bef916cf1